src/js/visual/configurator-csr.js

```diff
@@ -1333,6 +1354,166 @@ ripe.ConfiguratorCsr.prototype._onInitialsExtraEvent = function(self, initialsEx
+                lookAt: mayaScene.cameraLookAt
+            };
+        }
+
+        // unpacks config scene options
+        let zoomOptions;
+        if (config.scene) {
+            if (config.scene.zoom) {
+                zoomOptions = { ...config.scene.zoom };
+            }
+
+            // unpacks config scene camera options. It overrides the loaded scene values
+            if (config.scene.camera) {
+                cameraOptions.camera = { ...cameraOptions.camera, ...config.scene.camera };
+            }
+            if (config.scene.cameraLookAt) {
+                cameraOptions.cameraLookAt = {
+                    ...cameraOptions.cameraLookAt,
+                    ...config.scene.cameraLookAt
+                };
+            }
+        }
+
+        // unpacks config csr options
+        let rendererOptions;
+        const initialsOptions = {};
+        if (config.csr) {
+            if (config.csr.renderer) {
+                rendererOptions = { ...config.csr.renderer };
+            }
+
+            if (config.csr.personalization) {
+                initialsOptions.width = config.csr.personalization.width;
+                initialsOptions.height = config.csr.personalization.height;
+                initialsOptions.points = config.csr.personalization.points;
+                initialsOptions.position = config.csr.personalization.position;
+                initialsOptions.rotation = config.csr.personalization.rotation;
+                initialsOptions.scale = config.csr.personalization.scale;
+                initialsOptions.options = {
+                    textOptions: config.csr.personalization.text,
+                    materialOptions: config.csr.personalization.material,
+                    meshOptions: config.csr.personalization.mesh,
+                    baseTextureOptions: config.csr.personalization.baseTexture,
+                    displacementTextureOptions: config.csr.personalization.displacementTexture
+                };
+            }
+        }
+
+        this._initConfigDefaults({
+            rendererOptions: rendererOptions,
+            useDracoLoader: config.csr.useDracoLoader,
+            dracoLoaderDecoderPath: config.csr.dracoLoaderDecoderPath,
+            dracoLoaderDecoderFallbackPath: config.csr.dracoLoaderDecoderFallbackPath,
+            cameraOptions: cameraOptions,
+            zoomOptions: zoomOptions,
+            enabledInitials: initialsEnabled,
+            initialsOptions: initialsOptions
+        });
+
+        // init scene
+        this._initScene();
+
+        // init the CSR initials
+        this._initCsrRenderedInitials();
+
+        // init debug tools
+        this._initDebug();
+
+        // renders newly build scene
+        this._render();
+
+        self.loading = false;
+        this.trigger("ready");
+    };
+
+    // runs synchronously or asynchronously depending on how the CSR configurator was setup
+    this.awaitPostConfig ? await _postConfig() : _postConfig();

```

Comment:
what difference does this make for the user?

Reply:
It allows the loading to by asynchrounsly thus allowing blocking the thread